### PR TITLE
Support GHC 9.10

### PR DESCRIPTION
**Pull request by 9999years**
Depends on: https://github.com/google/ghc-source-gen/pull/117

### CODE REVIEW

**Comment by blackgnezdo** [comment]
Thanks for the fixes! Would you mind extending our CI build matrix to include 9.10. I'd be happy if you repurpose the cabal build with ghc 9.6 / cabal test for this. We already have 9.10 covered by Stack and I don't see ghc-9.10 in stackage yet.

File: .github/workflows/cabal-ci.yml [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
+name: cabal build
 jobs:
   runhaskell:
     name: cabal test
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version:
+          - ghc: 9.6.3
+            cabal: 3.10.2.0
```
Comment (blackgnezdo) [inline .github/workflows/cabal-ci.yml:13]:
```suggestion
```
We already have 9.6 covered by the stack matrix. Might as well not use it here. Could you remove `ghc96.cabal` files?

File: ghc9.6.3.cabal.project
Comment (blackgnezdo) [inline ghc9.6.3.cabal.project:1]: Please remove.

**Review by blackgnezdo — APPROVED** [review]
Looks good to me if everything passes and the file noted removed.

**Comment by blackgnezdo** [comment]
Thanks @9999years !

**Review by blackgnezdo — APPROVED** [review]
Looks great! Thanks again @9999years

**Comment by edsko** [comment]
Thanks for doing this! Any chance of a release? :)

**Comment by blackgnezdo** [comment]
I don't mind pushing the release button, but I'm not likely to find the time to do the version increment. We also got some awesome new features like ptorobuf editions support courtesy of @chungyc. I'd like somebody to note them down properly in the affected packages' changelogs.

**Comment by blackgnezdo** [comment]
Also, since we have a release of [ghc-source-gen  0.4.6.0](https://hackage.haskell.org/package/ghc-source-gen-0.4.6.0) up, we should be able to simplify this:
https://github.com/google/proto-lens/blob/c927e0341715a2ff7f87f219c9a36517f06cef80/ghc9.10.1.cabal.project#L6-L9

**Pull request merged**